components/DateTimePicker/TimePicker.tsx

```diff
@@ -0,0 +1,82 @@
+'use client';
+
+import { useState, useCallback } from 'react';
+
+import InputField from '@/components/InputField/InputField';
+
+/**
+ * TimePicker 컴포넌트는 시간을 선택할 수 있는 기능을 제공합니다.
+ * @param {string} width - optional, TimePicker 컴포넌트의 너비를 설정합니다. (기본 값은 w-full입니다.)
+ * @returns {JSX.Element} TimePicker 컴포넌트를 반환합니다.
+ */
+
+export default function TimePicker({ width = 'full' }: { width?: string }) {
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const [time, setTime] = useState<string>('');
+  const [amPm, setAmPm] = useState<'am' | 'pm'>('am');
+
+  const timeSlots = useCallback(() => {
+    const slots: string[] = [];
+    for (let i = 1; i <= 12; i++) {
+      slots.push(`${String(i)}:00`);
+      slots.push(`${String(i)}:30`);
+    }
+    return slots;
+  }, []);
+
+  return (
+    <>
+      <div
+        className={`flex flex-col gap-pr-8 ${width !== 'full' ? `w-pr-${width}` : 'w-full'}`}
+      >
+        <div className="relative">
+          <InputField
+            value={
+              time ? (amPm === 'am' ? `오전 ${time}` : `오후 ${time}`) : ''
+            }
+            placeholder="시간을 선택해주세요."
+            disabled={true}
+            onChange={() => {}}
+            name="date"
+          />
+          <button
+            className={`z-90 absolute left-0 top-0 size-full rounded-xl ${isOpen ? 'border border-brand-primary' : ''}`}
+            onClick={() => setIsOpen(!isOpen)}
+          />
+        </div>
+        {isOpen && (
+          <div className="flex gap-pr-14 rounded-xl border border-brand-primary bg-b-secondary p-pr-12">
+            <div className="flex flex-col gap-pr-8">
+              <button
+                onClick={() => setAmPm('am')}
+                className={`h-pr-40 w-pr-78 rounded-xl bg-b-primary text-14m text-t-default ${amPm === 'am' ? 'bg-brand-primary text-white' : ''}`}
+              >
+                오전
+              </button>
+              <button
+                onClick={() => setAmPm('pm')}
+                className={`h-pr-40 w-pr-78 rounded-xl bg-b-primary text-14m text-t-default ${amPm === 'pm' ? 'bg-brand-primary text-white' : ''}`}
+              >
+                오후
+              </button>
```

Comment:
위에 변수 객체와 classnames를 이용하면 이렇게 사용해볼 수 있을 것 같습니다~

```suggestion
              <button
                onClick={() => setAmPm('am')}
                className={getButtonClass('am')}
              >
                오전
              </button>
              <button
                onClick={() => setAmPm('pm')}
                className={getButtonClass('pm')}
              >
                오후
              </button>
```